DetectionResult[] results
bool is_green
